source/Databricks/source/SqlStatementExecution/Internal/DatabricksSqlStatementClient.cs

```diff
@@ -166,16 +191,39 @@ private async Task<SqlChunkResponse> GetChunkAsync(string chunkLink)
         return _responseResponseParser.ParseChunkResponse(jsonResponse);
     }
 
-    private async Task<TableChunk> GetChunkDataAsync(Uri? externalLink, string[] columnNames)
+    private async Task<TableChunk> GetChunkDataAsStringAsync(Uri? externalLink, string[] columnNames)
+    {
+        var httpResponse = await GetChunkDataAsync(externalLink);
+        var jsonResponse = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+        return _responseResponseParser.ParseChunkDataResponse(jsonResponse, columnNames);
+    }
+
+    private async Task<IAsyncEnumerable<string[]>> GetChunkDataAsStreamAsync(Uri? externalLink)
+    {
+        var httpResponse = await GetChunkDataAsync(externalLink);
+
+        var jsonResponse = await httpResponse.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        return _responseResponseParser.ParseChunkDataResponseAsync(jsonResponse);
+    }
+
+    private async Task<HttpResponseMessage> GetChunkDataAsync(Uri? externalLink)
     {
         var httpResponse = await _externalHttpClient.GetAsync(externalLink).ConfigureAwait(false);
         if (!httpResponse.IsSuccessStatusCode)
         {
-            throw new DatabricksSqlException($"Unable to get chunk data from external link {externalLink}. HTTP status code: {httpResponse.StatusCode}");
+            throw new DatabricksSqlException(
+                $"Unable to get chunk data from external link {externalLink}. HTTP status code: {httpResponse.StatusCode}");
         }
 
-        var jsonResponse = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
-        return _responseResponseParser.ParseChunkDataResponse(jsonResponse, columnNames);
+        return httpResponse;
+    }
+
+    private void LogDatabricksRequest(string sqlStatement, List<SqlStatementParameter> sqlStatementParameters)
+    {
+        _logger.LogDebug(
+            "Executing SQL statement: {Sql}, with parameters: {Parameters}",
+            HttpUtility.HtmlEncode(sqlStatement),
+            sqlStatementParameters);
```

Comment:
Nit: Only HtmlEncode sqlStatement if log level debug is enabled

```suggestion
        if (_logger.IsEnabled(LogLevel.Debug))
        {
            _logger.LogDebug(
                "Executing SQL statement: {Sql}, with parameters: {Parameters}",
                HttpUtility.HtmlEncode(sqlStatement),
                sqlStatementParameters);
        }
```